[CI] SimpleQueryStringBuilderTests.testToQuery failure

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/739/console

the following **does reproduce** on master:
```
./gradlew :server:test \
  -Dtests.seed=9F6E457C7C08D8E4 \
  -Dtests.class=org.elasticsearch.index.query.SimpleQueryStringBuilderTests \
  -Dtests.method="testToQuery" \
  -Dtests.security.manager=true \
  -Dtests.locale=lt \
  -Dtests.timezone=Africa/Luanda \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

```
  1> [2018-12-13T13:56:12,787][INFO ][o.e.p.PluginsService     ] [testToQuery] no plugins loaded
  1> [2018-12-13T13:56:13,559][INFO ][o.e.i.q.SimpleQueryStringBuilderTests] [testToQuery] after test
  2> REPRODUCE WITH: ./gradlew :server:test -Dtests.seed=9F6E457C7C08D8E4 -Dtests.class=org.elasticsearch.index.query.SimpleQueryStringBuilderTests -Dtests.method="testToQuery" -Dtests.security.manager=true -Dtests.locale=lt -Dtests.timezone=Africa/Luanda -Dcompiler.java=11 -Druntime.java=8
FAILURE 4.09s | SimpleQueryStringBuilderTests.testToQuery <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: query was marked as not cacheable in the context but this test indicates it should be cacheable: {
   >   "simple_query_string" : {
   >     "query" : "NOwvCZum",
   >     "analyzer" : "standard",
   >     "flags" : 589,
   >     "default_operator" : "and",
   >     "analyze_wildcard" : false,
   >     "minimum_should_match" : "75%",
   >     "auto_generate_synonyms_phrase_query" : false,
   >     "fuzzy_prefix_length" : 3,
   >     "fuzzy_max_expansions" : 50,
   >     "fuzzy_transpositions" : false,
   >     "boost" : 1.0,
   >     "_name" : "XPLcu0"
   >   }
   > }
   >    at __randomizedtesting.SeedInfo.seed([9F6E457C7C08D8E4:689547420D8B1D0E]:0)
   >    at org.elasticsearch.test.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:430)
   >    at java.lang.Thread.run(Thread.java:748)
```

--- a/server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java
+++ b/server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java
@@ -68,7 +68,8 @@ public class SimpleQueryStringBuilderTests extends AbstractQueryTestCase<SimpleQ
 
     @Override
     protected SimpleQueryStringBuilder doCreateTestQueryBuilder() {
-        SimpleQueryStringBuilder result = new SimpleQueryStringBuilder(randomAlphaOfLengthBetween(1, 10));
+        String queryText = randomAlphaOfLengthBetween(1, 10);
+        SimpleQueryStringBuilder result = new SimpleQueryStringBuilder(queryText);
         if (randomBoolean()) {
             result.analyzeWildcard(randomBoolean());
         }
@@ -105,9 +106,9 @@ public class SimpleQueryStringBuilderTests extends AbstractQueryTestCase<SimpleQ
                 fields.put(STRING_FIELD_NAME_2, 2.0f / randomIntBetween(1, 20));
             }
         }
-        // special handling if query is "now" and no field specified. This hits the "mapped_date" field which leads to the query not being
-        // cacheable and trigger later test failures (see https://github.com/elastic/elasticsearch/issues/35183)
-        if (fieldCount == 0 && result.value().equalsIgnoreCase("now")) {
+        // special handling if query start with "now" and no field specified. This hits the "mapped_date" field which leads to the query not
+        // being cacheable and trigger later test failures (see https://github.com/elastic/elasticsearch/issues/35183)
+        if (fieldCount == 0 && queryText.length() >= 3 && queryText.substring(0,3).equalsIgnoreCase("now")) {
             fields.put(STRING_FIELD_NAME_2, 2.0f / randomIntBetween(1, 20));
         }
 

